frontend/src/employee-frontend/components/unit/unit-reservations/ChildDateModal.tsx

```diff
@@ -207,6 +212,25 @@ export default React.memo(function ChildDateModal({
   const reservationElems = useFormElems(reservations)
   const attendanceElems = useFormElems(attendances)
 
+  const debouncedAttendances = useDebounce(
+    attendances.isValid() &&
+      attendances.value().every(({ endTime }) => endTime !== null)
+      ? attendances.value()
+      : null,
+    750
+  )
+  const expectedAbsencesEnabled =
+    date.isBefore(LocalDate.todayInHelsinkiTz()) &&
+    debouncedAttendances !== null
+  const expectedAbsences = useQueryResult(
+    childDateExpectedAbsencesQuery({
+      childId: child.id,
+      date,
+      attendances: debouncedAttendances ?? []
+    }),
+    { enabled: expectedAbsencesEnabled }
+  )
```

Comment:
This would enable you to use `renderResult`, because `expectedAbsences` would resolve to a `Success` of `null` when backend cannot be requested.

```suggestion
  const expectedAbsences = useQueryResult(
    queryOrDefault(
      (attendances: OpenTimeRange[]) =>
        childDateExpectedAbsencesQuery({
          childId: child.id,
          date,
          attendances
        }),
      null
    )(
      date.isBefore(LocalDate.todayInHelsinkiTz()) ? debouncedAttendances : null
    )
  )
```

Reply:
ah! 💡  I was longing for something like queryOrDefault, good to know it exists 😅

Reply:
Refactored like this https://github.com/espoon-voltti/evaka/pull/4610/commits/b2362a8f6f459c6fb60b0ae1d271d168c22dced3

I think `renderResult` is not necessarily useful here. The warnings are only soft info, so the user does not care that much whether it is still loading or even failed, and so the spinner would mostly just increase blinking without much value.